Bring up Ventura EWS Queues

chgibb-apple opened the pull request:
#### db2f07918fc08e3a41f990e4c097b1d75d395f06
<pre>
Bring up Ventura EWS Queues
<a href="https://rdar.apple.com/126490431">rdar://126490431</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=272686">https://bugs.webkit.org/show_bug.cgi?id=272686</a>

Reviewed by Jonathan Bedard.

* Tools/CISupport/ews-app/ews/common/github.py:
(GitHubEWS):
* Tools/CISupport/ews-app/ews/views/statusbubble.py:
(StatusBubble):
* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):

Canonical link: <a href="https://commits.webkit.org/277552@main">https://commits.webkit.org/277552@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6dd71a03189acd0d96ef89e425adfdffbfe408ac

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47845 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27039 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50670 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50527 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43899 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/50152 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/32912 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24528 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/38945 "Found 14 new test failures: imported/w3c/web-platform-tests/compat/webkit-box-rtl-flex.html, imported/w3c/web-platform-tests/css/css-pseudo/first-line-with-inline-block.html, imported/w3c/web-platform-tests/css/css-text-decor/text-emphasis-position-over-left-001.xht, imported/w3c/web-platform-tests/css/css-text-decor/text-emphasis-position-over-right-001.xht, imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-last-word-001.html, imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vertical-001.html, imported/w3c/web-platform-tests/css/css-text/text-encoding/shaping-no-join-001.html, imported/w3c/web-platform-tests/css/css-text/text-encoding/shaping-no-join-002.html, imported/w3c/web-platform-tests/css/css-text/text-encoding/shaping-no-join-003.html, imported/w3c/web-platform-tests/css/css-values/ch-unit-018.html ... (failure)") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48427 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/41368 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20248 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/42543 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5893 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/42972 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52421 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22884 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Sonoma-Debug-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/46251 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/47879 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24156 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45291 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10575 "") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/24947 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6799 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23876 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

briannafan commented on Tools/CISupport/ews-build/config.json:
```diff
@@ -227,28 +227,56 @@
+    },
+    {
+      "name": "macOS-Ventura-Release-Build-EWS", "shortname": "mac", "icon": "buildOnly",
+      "factory": "macOSBuildFactory", "platform": "mac-monterey",
```
Should this platform be `mac-ventura`? Same for `macOS-AppleSilicon-Ventura-Release-WK1-Tests-EWS`

chgibb-apple replied:
Yes it should. Thank you! Fixed.

chgibb-apple committed:
Bring up Ventura EWS Queues
rdar://126490431
https://bugs.webkit.org/show_bug.cgi?id=272686

Reviewed by Jonathan Bedard.

* Tools/CISupport/ews-app/ews/common/github.py:
(GitHubEWS):
* Tools/CISupport/ews-app/ews/views/statusbubble.py:
(StatusBubble):
* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):

Canonical link: https://commits.webkit.org/277552@main

aj062 commented on Tools/CISupport/ews-app/ews/common/github.py:
```diff
@@ -213,7 +213,13 @@ class GitHubEWS(GitHub):
                           ['merge', 'watch', '', 'jsc-armv7', ''],
-                          ['unsafe-merge', 'watch-sim', '', 'jsc-armv7-tests', '']]
+                          ['unsafe-merge', 'watch-sim', '', 'jsc-armv7-tests', ''],
+                          ['', '', 'mac-ventura', '', ''],
```
Each line here denotes one row in GitHub EWS status-bubble table.

So, all these new rows will have a single status-bubble, while there is space is rows above them. That would look somewhat ugly in the UI.
Some of these status-bubbles can be moved up in the table. For e.g.: 'mac-ventura' should be moved above in the line containing bubble for 'merge'. Similarly 'api-mac-ventura' and 'jsc-ventura-arm64' can be moved up. This would free up two additional rows.

aj062 commented on Tools/CISupport/ews-build/config.json:
```diff
@@ -310,7 +338,13 @@
+      "workernames": ["ews137"]
+    },
+    {
+      "name": "JSC-Tests-Ventura-arm64-EWS", "shortname": "jsc-ventura-arm64", "icon": "buildAndTest",
```
Since JSC queue is independent from other mac queues, it's possible to migrate this queue independently. Without the need for this long process of adding new queue, migrating machines and deleting old queue. Can just have a small PR to change the platform for `JSC-Tests-arm64-EWS` and update/move the two corresponding bots.

aj062 commented on Tools/CISupport/ews-build/config.json:
```diff
@@ -350,7 +384,13 @@
+      "workernames": ["ews100", "ews129", "ews142", "ews143", "ews150"]
+    },
+    {
+      "name": "API-Tests-macOS-AppleSilicon-Ventura-EWS", "shortname": "api-ventura-mac", "icon": "testOnly",
```
Now that most/all mac queues will be on AppleSilicon, we should consider dropping the AppleSilicon from queue names and adding intel/x86 wherever appropriate.
